tools/gluten-te/ubuntu/cbash-build.sh

```diff
@@ -56,4 +57,4 @@ CBASH_DOCKER_RUN_ARGS="$CBASH_DOCKER_RUN_ARGS $EXTRA_DOCKER_OPTIONS"
 CBASH_BASH_ARGS="$*"
 BASH_ARGS="$CBASH_BASH_ARGS"
 
-docker run $CBASH_DOCKER_RUN_ARGS $DOCKER_TARGET_IMAGE_BUILD_WITH_OS_IMAGE "cd /opt/gluten && $BASH_ARGS"
+docker run $CBASH_DOCKER_RUN_ARGS $DOCKER_PUSH_REGISTRY/$DOCKER_TARGET_IMAGE_BUILD_WITH_OS_IMAGE "cd /opt/gluten && $BASH_ARGS"
```

Comment:
Any reason making the change? Thanks!

Reply:
The format for the images' name is $DOCKER_PUSH_REGISTRY/$DOCKER_TARGET_IMAGE_BUILD_WITH_OS_IMAGE, rather than $DOCKER_TARGET_IMAGE_BUILD_WITH_OS_IMAGE. The prefix $DOCKER_PUSH_REGISTRY/ is necessary to correctly pull the images.

Reply:
Thanks and I understand your point. However I would be inclined to still use `$DOCKER_TARGET_IMAGE_BUILD_WITH_OS_IMAGE` rather than `$DOCKER_PUSH_REGISTRY/$DOCKER_TARGET_IMAGE_BUILD_WITH_OS_IMAGE` since the purpose of the code here was to locate and use the local image we just built within `build.sh`.

See code https://github.com/oap-project/gluten/blob/3fe4c959e07739ac142433485691f92a8a75bfd1/tools/gluten-te/ubuntu/build.sh#L73C51-L73C94

We used `-t $DOCKER_TARGET_IMAGE_BUILD_WITH_OS_IMAGE` to create a local image named `$DOCKER_TARGET_IMAGE_BUILD_WITH_OS_IMAGE`, during which we use `$DOCKER_PUSH_REGISTRY/$DOCKER_TARGET_IMAGE_BUILD_WITH_OS_IMAGE` as cache through command `--cache-from $DOCKER_CACHE_REGISTRY/$DOCKER_TARGET_IMAGE_BUILD_WITH_OS_IMAGE`.

So the pipeline here is, use remote image as cache, to build the local image.

The pros of this way, is when `DOCKER_CACHE_REGISTRY` is not set, the workflow / user could still be able to perform build to generate a local image `$DOCKER_TARGET_IMAGE_BUILD_WITH_OS_IMAGE` as if encountering a cache miss.

As an experiment, you can check whether your new way in the patch could be able to handle the case that `DOCKER_CACHE_REGISTRY` is not set.

Reply:
You are right, there should not be any changes made here. It has been modified.